packages/core/lib/datasources/Datasource.js

```diff
@@ -146,11 +149,21 @@ class Datasource extends EventEmitter {
     onError && outputQuads.on('error', onError);
 
     // Execute the query
+    query = this._convertToTerms(query);
+
     try { this._executeQuery(query, destination); }
     catch (error) { outputQuads.emit('error', error); }
     return outputQuads;
   }
 
+  _convertToTerms(query) {
```

Comment:
The string values still have to be converted to rdfjs term though?

Reply:
Yes indeed, they still have to be converted. But Datasource doesn't feel like a logical place to do that.
Intuitively, I would suspect that this would be done in `QuadPatternRouter`. What do you think?